pkg/cloud/openstack/services/networking/loadbalancer.go

```diff
@@ -0,0 +1,437 @@
+			listenerCreateOpts := listeners.CreateOpts{
+				Name:           lbPortObjectsName,
+				Protocol:       "TCP",
+				ProtocolPort:   port,
+				LoadbalancerID: lb.ID,
+			}
+			listener, err = listeners.Create(s.client, listenerCreateOpts).Extract()
+			if err != nil {
+				return fmt.Errorf("error creating listener: %s", err)
+			}
+			err = waitForLoadBalancer(s.client, lb.ID, "ACTIVE")
+			if err != nil {
+				return err
+			}
+			err = waitForListener(s.client, listener.ID, "ACTIVE")
+			if err != nil {
+				return err
+			}
+		}
+
+		// lb pool
+		pool, err := checkIfPoolExists(s.client, lbPortObjectsName)
+		if err != nil {
+			return err
+		}
+		if pool == nil {
+			klog.Infof("Creating lb pool %s", lbPortObjectsName)
+			poolCreateOpts := pools.CreateOpts{
+				Name:       lbPortObjectsName,
+				Protocol:   "TCP",
+				LBMethod:   pools.LBMethodRoundRobin,
+				ListenerID: listener.ID,
+			}
+			pool, err = pools.Create(s.client, poolCreateOpts).Extract()
+			if err != nil {
+				return fmt.Errorf("error creating pool: %s", err)
+			}
+			err = waitForLoadBalancer(s.client, lb.ID, "ACTIVE")
+			if err != nil {
+				return err
+			}
+		}
+
+		// lb monitor
+		monitor, err := checkIfMonitorExists(s.client, lbPortObjectsName)
+		if err != nil {
+			return err
+		}
+		if monitor == nil {
+			klog.Infof("Creating lb monitor %s", lbPortObjectsName)
+			monitorCreateOpts := monitors.CreateOpts{
+				Name:       lbPortObjectsName,
+				PoolID:     pool.ID,
+				Type:       "TCP",
+				Delay:      30,
+				Timeout:    5,
+				MaxRetries: 3,
+			}
+			_, err = monitors.Create(s.client, monitorCreateOpts).Extract()
+			if err != nil {
+				return fmt.Errorf("error creating monitor: %s", err)
+			}
+			err = waitForLoadBalancer(s.client, lb.ID, "ACTIVE")
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	clusterProviderStatus.Network.APIServerLoadBalancer = &providerv1.LoadBalancer{
+		Name:       lb.Name,
+		ID:         lb.ID,
+		InternalIP: lb.VipAddress,
+		IP:         fp.FloatingIP,
+	}
+	return nil
+}
+
+func (s *Service) ReconcileLoadBalancerMember(clusterName string, machine *clusterv1.Machine, clusterProviderStatus *providerv1.OpenstackClusterProviderStatus) error {
+
+	if !util.IsControlPlaneMachine(machine) {
+		return nil
+	}
+
+	if clusterProviderStatus.Network == nil {
+		return errors.New("network is not yet available in clusterProviderStatus")
+	}
+	if clusterProviderStatus.Network.Subnet == nil {
+		return errors.New("network.Subnet is not yet available in clusterProviderStatus")
+	}
+	if clusterProviderStatus.Network.APIServerLoadBalancer == nil {
+		return errors.New("network.APIServerLoadBalancer is not yet available in clusterProviderStatus")
+	}
+
+	loadBalancerName := fmt.Sprintf("%s-cluster-%s-%s", networkPrefix, clusterName, kubeapiLBSuffix)
+
+	lbID := clusterProviderStatus.Network.APIServerLoadBalancer.ID
+	subnetID := clusterProviderStatus.Network.Subnet.ID
+
+	for _, port := range []int{22, 6443} {
```

Comment:
Maybe make these constants so there's no magic numbers even if it's apparent what they're for. I'd say 6443 is less obvious if someone is unfamiliar with CAPI.

Reply:
Oh I actually added properties for this, just forgot to use them. Thx :)

Reply:
So I will iterate over: APIServerLoadBalancerPort  & APIServerLoadBalancerAdditionalPorts

Reply:
fixed